habitat-baselines/habitat_baselines/rl/hrl/hrl_ppo.py

```diff
@@ -0,0 +1,128 @@
+import torch
+import torch.nn.functional as F
+
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.rl.ddppo.algo.ddppo import DecentralizedDistributedMixin
+from habitat_baselines.rl.ppo import PPO
+from habitat_baselines.utils.common import (
+    LagrangeInequalityCoefficient,
+    inference_mode,
+)
+
+
+@baseline_registry.register_updater
+class HrlPPO(PPO):
+    def _update_from_batch(self, batch, epoch, rollouts, learner_metrics):
+        n_samples = max(batch["loss_mask"].sum(), 1)
+
+        def record_min_mean_max(t: torch.Tensor, prefix: str):
+            for name, op in (
+                ("min", torch.min),
+                ("mean", torch.mean),
+                ("max", torch.max),
+            ):
+                learner_metrics[f"{prefix}_{name}"].append(op(t))
+
+        def reduce_loss(loss):
+            return (loss * batch["loss_mask"]).sum() / n_samples
+
+        self._set_grads_to_none()
+
+        (
+            values,
+            action_log_probs,
+            dist_entropy,
+            _,
+            aux_loss_res,
+        ) = self._evaluate_actions(
+            batch["observations"],
+            batch["recurrent_hidden_states"],
+            batch["prev_actions"],
+            batch["masks"],
+            batch["actions"],
+            batch["rnn_build_seq_info"],
+        )
+
+        ratio = torch.exp(action_log_probs - batch["action_log_probs"])
+
+        surr1 = batch["advantages"] * ratio
```

Comment:
I fixed the capitalization to be consistent.

I agree there is code duplication, but de-duplicating will be difficult and I think is out of scope for this PR since this will make verifying training more difficult.